include/seqan3/alignment/matrix/alignment_matrix_formatter.hpp

```diff
@@ -0,0 +1,372 @@
+ *
+ *
+ * ## Example
+ *
+ * \snippet test/snippet/alignment/matrix/alignment_score_matrix.cpp code
+ *
+ * ### Output
+ * \include test/snippet/alignment/matrix/alignment_score_matrix.out
+ */
+template <alignment_matrix_concept alignment_matrix_t>
+struct alignment_matrix_formatter
+{
+    //!\brief The type of the #matrix
+    using alignment_matrix_type = alignment_matrix_t;
+
+    //!\brief The matrix to format.
+    alignment_matrix_type const & matrix;
+
+    //!\brief The actual format used by the formatter.
+    alignment_matrix_format symbols{alignment_matrix_format::unicode_arrows};
+
+    //!\brief Whether #alignment_matrix_type is a traceback matrix.
+    //!\hideinitializer
+    static constexpr bool is_traceback_matrix = []
+    {
+        using trace_matrix_t = detail::transfer_template_args_onto_t<alignment_matrix_type, alignment_trace_matrix>;
+        return std::is_same_v<alignment_matrix_type, trace_matrix_t>;
+    }();
+
+    //!\brief Determines the largest width of all entries in the #matrix,
+    //!       e.g. `-152` has width 4.
+    std::size_t auto_width() const noexcept
+    {
+        std::size_t col_width = 1;
+        for(unsigned row = 0; row < matrix.rows(); ++row)
+            for(unsigned col = 0; col < matrix.cols(); ++col)
+                col_width = std::max(col_width, char_length(char_at(row, col)));
+        return col_width;
+    }
+
+    //!\brief Print the formatted #matrix to std::cout.
+    //!\param[in]  column_width    width of each cell, std::nullopt defaults to auto_width()
+    void format(std::optional<std::size_t> column_width = std::nullopt) const noexcept
+    {
+        format(std::cout, column_width);
+    }
+
+    /*!\brief Print the formatted #matrix to the \a cout stream.
+     * \tparam         char_t          char type of std::basic_ostream
+     * \tparam         traits_t        traits type of std::basic_ostream
+     * \param[in,out]  cout            print formatted #matrix into this ostream
+     * \param[in]      column_width    width of each cell, std::nullopt defaults to auto_width()
+     */
+    template <typename char_t, typename traits_t>
+    void format(std::basic_ostream<char_t, traits_t> & cout, std::optional<std::size_t> column_width) const noexcept
+    {
+        std::size_t _column_width = column_width.has_value() ? column_width.value() : auto_width();
+
+        auto cell = [&](auto && symbol, std::optional<std::size_t> symbol_bytes = std::nullopt)
+        {
+            // deal with unicode chars that mess up std::setw
+            std::size_t bytes = symbol_bytes.value_or(
+                is_traceback_matrix ? symbols.trace_dir_bytes : 1u);
+            std::size_t length_bytes = char_length_bytes(symbol);
+            std::size_t length = length_bytes / bytes;
+            std::size_t offset = length_bytes - length;
+
+            cout << std::left
+                      << std::setw(_column_width + offset)
```

Comment:
usually the operator is added to the end of a line before the line wrap, and the following line is aligned to the expression above 💅